test/integration/seaport/seaport-integration.js

```diff
@@ -0,0 +1,205 @@
+
+    seaportFixtures = await seaportFixtures(seaport);
+
+    // Deploy diamond
+    let [protocolDiamond, , , , accessController] = await deployProtocolDiamond(maxPriorityFeePerGas);
+
+    // Cast Diamond to contract interfaces
+    const offerHandler = await ethers.getContractAt("IBosonOfferHandler", protocolDiamond.address);
+    const accountHandler = await ethers.getContractAt("IBosonAccountHandler", protocolDiamond.address);
+    const fundsHandler = await ethers.getContractAt("IBosonFundsHandler", protocolDiamond.address);
+
+    // Grant roles
+    await accessController.grantRole(Role.PROTOCOL, protocol.address);
+    await accessController.grantRole(Role.PROTOCOL, protocolDiamond.address);
+    await accessController.grantRole(Role.UPGRADER, deployer.address);
+
+    const protocolClientArgs = [protocolDiamond.address];
+
+    const [, beacons, proxies, bv] = await deployProtocolClients(protocolClientArgs, maxPriorityFeePerGas);
+
+    [bosonVoucher] = bv;
+    const [beacon] = beacons;
+    const [proxy] = proxies;
+
+    const protocolFeeFlatBoson = ethers.utils.parseUnits("0.01", "ether").toString();
+    const buyerEscalationDepositPercentage = "1000"; // 10%
+
+    [bosonToken] = await deployMockTokens();
+
+    // Add config Handler, so ids start at 1, and so voucher address can be found
+    const protocolConfig = [
+      // Protocol addresses
+      {
+        treasury: protocolTreasury.address,
+        token: bosonToken.address,
+        voucherBeacon: beacon.address,
+        beaconProxy: proxy.address,
+      },
+      // Protocol limits
+      {
+        maxExchangesPerBatch: 100,
+        maxOffersPerGroup: 100,
+        maxTwinsPerBundle: 100,
+        maxOffersPerBundle: 100,
+        maxOffersPerBatch: 100,
+        maxTokensPerWithdrawal: 100,
+        maxFeesPerDisputeResolver: 100,
+        maxEscalationResponsePeriod: oneMonth,
+        maxDisputesPerBatch: 100,
+        maxAllowedSellers: 100,
+        maxTotalOfferFeePercentage: 4000, //40%
+        maxRoyaltyPecentage: 1000, //10%
+        maxResolutionPeriod: oneMonth,
+        minDisputePeriod: oneWeek,
+        maxPremintedVouchers: 10000,
+      },
+      //Protocol fees
+      {
+        percentage: 200, // 2%
+        flatBoson: protocolFeeFlatBoson,
+        buyerEscalationDepositPercentage,
+      },
+    ];
+
+    const facetNames = [
+      "ExchangeHandlerFacet",
+      "OfferHandlerFacet",
+      "SellerHandlerFacet",
+      "DisputeResolverHandlerFacet",
+      "FundsHandlerFacet",
+      "ProtocolInitializationHandlerFacet",
+      "ConfigHandlerFacet",
+    ];
+
+    const facetsToDeploy = await getFacetsWithArgs(facetNames, protocolConfig);
+
+    // Cut the protocol handler facets into the Diamond
+    await deployAndCutFacets(protocolDiamond.address, facetsToDeploy, maxPriorityFeePerGas);
+
+    const seller = mockSeller(assistant.address, assistant.address, assistant.address, assistant.address);
```

Comment:
Code for protocol deployment and test preparation may be factorized for all integration test (for instance in ./test/integration/utils.ts?), in order to avoid redundancies and ensure more rational maintenance effort.
(for example, very similar code can be found in `./test/integration/02-Upgraded-facet.js`)

Reply:
Yes, we need to refactor those before/beforeEach code block on the tests. I have already added a task for this https://app.asana.com/0/1200973286514456/1204060145333998